subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -449,3 +449,14 @@ If an <<#sec:configuring_toolchain_specifications, unconfigured>> toolchain spec
 This can be achieved by passing an empty lambda when requesting a tool: `javaToolchainService.launcherFor({})`.
 
 You can find more details on defining custom tasks in the <<more_about_tasks.adoc#more_about_tasks, Authoring tasks>> documentation.
+
+[[sec:limitations]]
+== Toolchains limitations
+
+Gradle may detect toolchains incorrectly when the build is run in a JVM/JRE compiled against `musl`, https://musl.libc.org/[an alternative implementation] of the C standard library.
+A JVM/JRE compiled against `musl` may override the `LD_LIBRARY_PATH` environment variable in order to control dynamic library resolution.
+This might affect java subprocesses launched by Gradle, resulting in unexpected behavior.
+
+Thus, usage of multiple java toolchains is discouraged in environments with the `musl` library.
```

Comment:
```suggestion
As a consequence, using multiple different toolchains in a build is discouraged in environments based on the `musl` library.
```

Reply:
This is not clear enough. What exactly is the non-recommended scenario? Running Gradle with one Java version, having a toolchain configured with another version? Having two different toolchains configured in the build with two Java versions? Please clarify, so that whoever reads this knows exactly what's ok to do and what not.

Reply:
`environments based on` doesn't sound right to me. Ubuntu is not based on musl, but you can have a JVM compiled against musl there.

The same goes for `in build` – it's not just building, any java subprocess can be affected.  

As for clarity - I guess the last sentence clears this, doesn't it?
`If you are using a single toolchain to build and run your application, you can safely ignore this limitation.`